src/pages/shoppingListsPage/shoppingListsPage.test.tsx

```diff
@@ -1090,4 +1089,139 @@ describe('ShoppingListsPage', () => {
       })
     })
   })
+
+  describe('destroying a list item', () => {
+    describe('when the user cancels', () => {
+      const mockServer = setupServer(getShoppingListsSuccess)
+
+      beforeAll(() => mockServer.listen())
+      beforeEach(() => mockServer.resetHandlers())
+      afterAll(() => mockServer.close())
+
+      test("doesn't remove the item", async () => {
+        const wrapper = renderAuthenticated(
+          <PageProvider>
+            <GamesContext.Provider value={gamesContextValue}>
+              <ShoppingListsProvider>
+                <ShoppingListsPage />
+              </ShoppingListsProvider>
+            </GamesContext.Provider>
+          </PageProvider>
+        )
+
+        window.confirm = vitest.fn().mockImplementation(() => false)
+
+        wrapper.debug()
+        const destroyIcon = await wrapper.findByTestId(
+          'destroyShoppingListItem3'
+        )
+
+        act(() => {
+          fireEvent.click(destroyIcon)
+        })
+
+        await waitFor(() => {
+          expect(window.confirm).toHaveBeenCalled()
+          expect(wrapper.getAllByText('Iron ingot').length).toEqual(2)
+          expect(
+            wrapper.getByText(
+              'OK, your shopping list item will not be deleted.'
+            )
+          ).toBeTruthy()
+        })
+      })
+    })
+
+    describe('when successful', () => {
+      const mockServer = setupServer(
+        getShoppingListsSuccess,
+        deleteShoppingListItemSuccess
+      )
+
+      beforeAll(() => mockServer.listen())
+      beforeEach(() => mockServer.resetHandlers())
+      afterAll(() => mockServer.close())
+
+      test('removes the item from the regular and aggregate list', async () => {
+        const wrapper = renderAuthenticated(
+          <PageProvider>
+            <GamesContext.Provider value={gamesContextValue}>
+              <ShoppingListsProvider>
+                <ShoppingListsPage />
+              </ShoppingListsProvider>
+            </GamesContext.Provider>
+          </PageProvider>
+        )
+
+        window.confirm = vitest.fn().mockImplementation(() => true)
+
+        const destroyIcon = await wrapper.findByTestId(
+          'destroyShoppingListItem3'
+        )
+
+        act(() => {
+          fireEvent.click(destroyIcon)
+        })
+
+        await waitFor(() => {
+          expect(window.confirm).toHaveBeenCalled()
+
+          // Should destroy both list items
+          expect(wrapper.queryByText('Iron ingot')).toBeFalsy()
+
+          // Should not remove any other list items
+          expect(wrapper.getAllByText('Necklace').length).toEqual(2)
```

Comment:
This kind of tests the MSW handler but it also tests that there isn't front-end logic removing items not removed at the API.